### PR TITLE
[WIP] Single+3 point slings

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
SUMMARY: Content "Adds variations of the shoulder strap"
#### Purpose of change

Currently, the only options for toting around a long arm are using a sling or a holster of some sort. Some have expressed interest in different types of slings ( #29571 :the survivor harness, minus its storage bits), and as such I've drawn up what I believe are two tactically useful variations of this.

#### Describe the solution

Two additional sling types have been added: 

- The single point sling: very, very fast to draw, put on and take off, but incredibly encumbering. This is most useful for a survivor's 'Truck Gun', something that can very quickly be stowed or readied. As it not exactly worn, but rather is just attached to you, it's very encumbering.  

- The three-point sling: Slightly less encumbering than a traditional 2 point because it doesn't have the latter's tendency to slide around as much. As long as your chest is not overly encumbered and there's nothing for it to snag on, it's not too slow to draw. Take a bit more time to wear than other slings. [Here's Ian McCollum demonstrating how to use a 3 point.](https://www.full30.com/watch/MDA2MDQ2/rifle-slings---the-three-point-sling)  [And here's how they can seriously fail.](https://www.youtube.com/watch?v=HwSXvNcbOv0) 

Both factor in firearm encumbrance in different ways, and have different wear/remove speeds. Both are in some ways better than a two point sling. And both have very serious drawbacks compared to the traditional sling.
There's no free lunch.

#### Describe alternatives you've considered
Converting both slings into different types of 'holsters'.
#### Additional context

TODO:

- Balance:
The values will need playing with for balance. I'm shooting for the single point being practical for short term use with a subgun(~15-20 encumb), and debilitating with a large rifle (~40ish).  The three point needs to be slightly slower than a 2 point unless you're wearing anything more than a medium weight jacket (say, a trench coat?), in which case it'd take ~1/2 minute or more to get off/wield/wear. 

- Adjust when the encumbrance while containing function assigns encumbrance to guns
Ideally this will make fix the belt clip and give it the same encumbrance calculations. I've already gotten it to work with these two items.

- Figure out why this isn't working with wielding
Wearing/removing seem to be working correctly, but wielding from worn doesn't seem to be following the same calculations.

- probably astyle the crap out of every c++ function I touch

### CODE REVIEW

**Comment by Rail-Runner** [comment]
I really think different sling effects should be JSONized somehow, at the very least into an item flag, and at best, a separate JSON field. Item ID hardcoding is a bad practice.

**Comment by tenmillimaster** [comment]
Done. I've removed the hardcoded ID's (the normal sling's id too).

File: src/player.cpp [hunk 1]
```diff
@@ -8216,7 +8216,11 @@ bool player::wield( item &target )
 
     bool worn = is_worn( target );
     int mv = item_handling_cost( target, true,
-                                 worn ? INVENTORY_HANDLING_PENALTY / 2 : INVENTORY_HANDLING_PENALTY );
+                                if( worn ) {
```
Comment (ZhilkinSerg) [inline src/player.cpp]: That is invalid syntax. You probably want something like this:

```C++
    bool worn = is_worn( target );
    int base_cost_mult = 1;
    if( worn ) {
        if( target.has_flag( "SLUNG1" ) ) {
            base_cost_mult = 0.01 * std::max( target.get_encumber( *this ) / 10.0, 1.0 );
        } else if( target.has_flag( "SLUNG3" ) ) {
            base_cost_mult = 3 * std::max( encumb( bp_torso ) / 20, 1 );
        } else {
            base_cost_mult = 0.5;
        }
    }
    int mv = item_handling_cost( target, true, INVENTORY_HANDLING_PENALTY * base_cost_mult );
```

**Comment by ZhilkinSerg** [comment]
Marking as Stalled. Please ping me to reopen when you are ready to fix the code.

**Pull request closed without merging**